### PR TITLE
67033: map military/int'l address elements to single LH address

### PR DESCRIPTION
**Pull request by sethdarragile6**
## Summary

- This modifies the `Form526ToLighthouseTransform` class to map military or international address elements to a single Lighthouse address in the 526 /submit endpoint request.
- Team: DBex Trex 🦖 (Disability Benefits Experience 1)
- The `Form526ToLighthouseTransform` class is currently not in use. When 526 /submit migration is implemented, it will have its own dedicated flipper switch.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#67033
- [link](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/67033)

## Testing done

- extending existing tests to test for the new mapping functionality

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 1]
```diff
@@ -258,9 +260,13 @@ def transform_mailing_address(veteran, veteran_identification)
...
-        veteran_identification.mailing_address.city = veteran['currentMailingAddress']['city']
-        veteran_identification.mailing_address.state = veteran['currentMailingAddress']['state']
-        veteran_identification.mailing_address.zip_first_five = veteran['currentMailingAddress']['zipFirstFive']
+        veteran_identification.mailing_address.city = veteran['currentMailingAddress']['militaryPostOfficeTypeCode'] ||
+                                                      veteran['currentMailingAddress']['city']
+        veteran_identification.mailing_address.state = veteran['currentMailingAddress']['militaryStateCode'] ||
+                                                       veteran['currentMailingAddress']['state']
+        veteran_identification.mailing_address.zip_first_five =
+          veteran['currentMailingAddress']['internationalPostalCode'] ||
+          veteran['currentMailingAddress']['zipFirstFive']
```
Comment (mchae-nava) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:269]: Excuse my ignorance. Just double checking, is there a possibility where a Veteran could potentially list both a military address and a civilian address? 

And if so, do we have an official way to know which takes priority? (e.g. military always priority, most recent takes priority, etc.)

File: spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb [hunk 2]
```diff
@@ -114,6 +125,18 @@
...
+    it 'sets military/intl address correctly' do
+      data['form526']['veteran']['changeOfAddress']['militaryPostOfficeTypeCode'] = 'APO'
+      data['form526']['veteran']['changeOfAddress']['militaryStateCode'] = 'AE'
+      data['form526']['veteran']['changeOfAddress']['internationalPostalCode'] = '817'
+
+      result = transformer.transform_change_of_address(data['form526']['veteran'])
+      expect(result.city).to eq('APO')
+      expect(result.state).to eq('AE')
+      expect(result.zip_first_five).to eq('817')
+      expect(result.dates).not_to be_nil
```
Comment (mchae-nava) [inline spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb:138]: Do we have unit tests for negative scenarios? Like if the data provided is null or not usable for some reason. And if the priority consideration ends up being a thing, tests for that would be good as wel

**Review by mchae-nava — COMMENTED** [review]
Looks mostly good to me, just some questions for my own lack of context

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb (hunk 1 above)
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:269]: It is _not_ possible. What gets submitted is one address (current mailing address and/or  change of address), and the transform code in data_transformation_all_claims adds new military/intl elements to the JSON based on the state and country submitted. We send these to EVSS as-is, but as it turns out (per Michael Harlow) EVSS is simply filling those elements into one address, anyway. So for LH, we're just doing that on _our_ end now. If militaryPostOfficeTypeCode, use it for city instead, and so for militaryStateCode + state and internationalPostalCode + zipFirstFive

File: spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb (hunk 2 above)
Comment (sethdarragile6) [reply spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb:138]: You can't see it fully in this view, but the negative scenario is basically covered by the test right above it

**Review by aurora-a-k-a-lightning — APPROVED** [review]
lgtm

**Review by mchae-nava — APPROVED** [review]
Also LGTM! Thanks for the clarifications

**Pull request merged**